File: Dev/asd_cs/Particular/Lambda.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace asd.Particular
{
	internal static class Lambda
	{
		static BucketSort<Object2D> updateObject2DSorter = new BucketSort<Object2D>((_)=> _.UpdatePriority);
		static BucketSort<Object2D> drawObject2DSorter = new BucketSort<Object2D>((_) => _.UpdatePriority);
		static BucketSort<Layer> updateLayerSorter = new BucketSort<Layer>((_) => _.UpdatePriority);
		static BucketSort<Layer> drawLayerSorter = new BucketSort<Layer>((_) => _.UpdatePriority);

		public static bool HasContentHavingSpecificUpdatePriority<TContent>(
			SortedList<int, LinkedList<TContent>> contents,
			int updatePriority)
		{
			return contents.Any(x => x.Key == updatePriority);
		}

		public static IEnumerable<Object2D> FilterDeadObject(List<Object2D> objects)
		{
			return objects.Where(x => x.IsAlive);
		}

		public static void RemoveDead(List<Object2D> objects)
		{
			objects.RemoveAll(x => !x.IsAlive);
		}

		public static void SortByUpdatePriority(List<Layer> layers)
		{
			updateLayerSorter.Sort(layers);
		}

		public static IEnumerable<TContent> ToLinear<TContent>(SortedList<int, LinkedList<TContent>> contents)
		{
			return contents.SelectMany(x => x.Value);
		}

		public static void SortByDrawingPriority(List<Layer> layersToDraw)
		{
			drawLayerSorter.Sort(layersToDraw);
		}

		class BucketSort<T>
		{
			Func<T, int> getPriority = null;
			SortedDictionary<int, List<T>> tempObjects = new SortedDictionary<int, List<T>>();

			public BucketSort(Func<T, int> getPriorityFunc)
			{
				getPriority = getPriorityFunc;
			}
			public void Sort(List<T> objects)
			{
				foreach(var o in objects)
				{
					var priority = getPriority(o);
					if(!tempObjects.ContainsKey(priority))
					{
						tempObjects.Add(priority, new List<T>());
					}
					tempObjects[priority].Add(o);
				}

				objects.Clear();

				foreach(var to in tempObjects)
				{
					objects.AddRange(to.Value);
					to.Value.Clear();
				}
			}
		}
	}
}

File: Dev/asd_java/Particular/Lambda.java
package asd.Particular;

import java.util.Iterator;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

public class Lambda
{
	public static <TContent> boolean HasContentHavingSpecificUpdatePriority(
		java.util.TreeMap<Integer, java.util.LinkedList<TContent>> contents,
		int updatePriority)
	{
		Iterator<Integer> it = contents.keySet().iterator();
        while (it.hasNext())
		{
			Integer key = it.next();
			if(key == updatePriority) return true;
        }

		return false;

		//return contents.Any(x => x.Key == updatePriority);
	}

	public static java.lang.Iterable<asd.Object2D> FilterDeadObject(java.util.ArrayList<asd.Object2D> objects)
	{
		return objects.stream().filter(x -> x.getIsAlive())::iterator;
	}

	public static void RemoveDead(java.util.ArrayList<asd.Object2D> objects)
	{
		java.util.ArrayList<asd.Object2D> removed = new java.util.ArrayList<asd.Object2D>();
		for(asd.Object2D o : objects)
		{
			if(!o.getIsAlive())
			{
				removed.add(o);
			}
		}
		objects.removeAll(removed);
	}

	public static void SortByUpdatePriority(java.util.ArrayList<asd.Layer> layers)
	{
		Collections.sort(layers,
			new Comparator<asd.Layer>()
			{
				public int compare( asd.Layer x, asd.Layer y ){ return x.getUpdatePriority() - y.getUpdatePriority(); }
			}
		);

		//layers.Sort((x, y) -> x.UpdatePriority - y.UpdatePriority);
	}

	public static <TContent> java.lang.Iterable<TContent> ToLinear(java.util.TreeMap<Integer, java.util.LinkedList<TContent>> contents)
	{
		return contents.values().stream().flatMap(v->v.stream())::iterator;
		//return contents.SelectMany(x -> x.Value);
	}

	public static void SortByDrawingPriority(java.util.ArrayList<asd.Layer> layersToDraw)
	{
		Collections.sort(layersToDraw,
			new Comparator<asd.Layer>()
			{
				public int compare( asd.Layer x, asd.Layer y ){ return x.getDrawingPriority() - y.getDrawingPriority(); }
			}
		);

		//layersToDraw.Sort((x, y) -> x.DrawingPriority - y.DrawingPriority);
	}
}
